build_config/README.md

```diff
@@ -54,6 +54,26 @@ configuration may have the following keys:
   on the particular builder. Values in this map will override the default
   provided by builder authors. Values may also be overridden based on the build
   mode with `dev_options` or `release_options`.
+- **dev_options**: Map, Optional: A free-form map which will be passed to the
+  `Builder` as a `BuilderOptions` when it is constructed. Usage varies depending
+  on the particular builder. The values in this map override Builder defaults or
+  non mode-specific options per-key when the build is done in dev mode.
+- **release_options**: Map, Optional: A free-form map which will be passed to
+  the `Builder` as a `BuilderOptions` when it is constructed. Usage varies
+  depending on the particular builder. The values in this map override Builder
+  defaults or non mode-specific options when the build is done in release mode.
+
+## Configuring `Builder`s globally
```

Comment:
Discussed offline. It would be useful to have some sample configuration or use case for the global_options.

Reply:
yeah I think that will come once we get to https://github.com/dart-lang/build/issues/1308#issuecomment-388938427 we'll add an example to our docs which uses global options